layers/utils.py

```diff
@@ -0,0 +1,72 @@
+import json
+import os
+import subprocess
+
+
+def run(args):
+    print(">", " ".join(args))
+    subprocess.run(args, check=True)
+
+
+def ensureEmptyTempDirectoryExists(directoryName):
+    if os.path.isdir(directoryName):
+        run(["rm", "-r", directoryName])
+    os.makedirs(directoryName, exist_ok=True)
+
+
+def convertPbfToGeoJson(pbfPath, geojsonPath, geometryType):
+    run(
+        [
+            "osmium",
+            "export",
+            pbfPath,
+            f"--geometry-type={geometryType}",
+            "-o",
+            geojsonPath,
+        ]
+    )
+
+
+# Note the layer name is based on the input filename. This always generates
+# numeric feature IDs.
+def convertGeoJsonToPmtiles(geojsonPath, pmtilesPath):
+    run(["tippecanoe", geojsonPath, "--generate-ids", "-o", pmtilesPath])
+
+
+# Adds numeric IDs to every feature, trims coordinate precision, and uses the
+# callback to transform each feature's properties. Overwrites the file.
+#
+# The callback takes (input properties, output properties) and doesn't return
+# anything. Output is a blank dictionary that should be filled out.
+def cleanUpGeojson(path, transformProperties):
+    print(f"Cleaning up {path}")
+    gj = {}
+    with open(path) as f:
+        gj = json.load(f)
+
+        counter = 1
+        for feature in gj["features"]:
+            properties = {}
+            transformProperties(feature["properties"], properties)
```

Comment:
Might be nicer to enforce that `transformProperties` returns the object.  Not only is the line below (51) kinda weird, but if we wanted to use any functional style map it would expect to do it this way.

Reply:
Done. It doesn't make sense to pass in a blank {} then, so also changed to have each place define it. I was trying to be more minimalist (imagine all these lines in a lambda expression), but it's not so bad as a method